it/src/test/java/org/corfudb/universe/scenario/RotateLinkFailureIT.java

```diff
@@ -0,0 +1,141 @@
+
+                String firstServerToKick = Collections.max(Arrays.asList(server0.getEndpoint(), server1.getEndpoint()));
+                assertThat(corfuClient.getLayout().getUnresponsiveServers()).containsExactly(firstServerToKick);
+
+                Layout lastLayout = corfuClient.getLayout();
+
+                // Verify data path working fine
+                for (int i = 0; i < DEFAULT_TABLE_ITER; i++) {
+                    assertThat(table.get(String.valueOf(i))).isEqualTo(String.valueOf(i));
+                }
+
+                // 2nd link failure rotation, disconnect between server1 and server2
+                // and heal previous link failure between server0 and server1
+                server1.disconnect(Collections.singletonList(server2));
+                server0.reconnect(Collections.singletonList(server1));
+
+                // Wait for some time to ensure cluster stabilizes
+                // Server1 should stay in unresponsive set, no layout change
+                waitUninterruptibly(20);
+                assertThat(corfuClient.getLayout()).isEqualTo(lastLayout);
+
+                // Verify data path working fine
+                for (int i = 0; i < DEFAULT_TABLE_ITER; i++) {
+                    assertThat(table.get(String.valueOf(i))).isEqualTo(String.valueOf(i));
+                }
+
+                // 3rd link failure rotation, disconnect between server2 and server0
+                // and heal previous link failure between server1 and server2
+                server2.disconnect(Collections.singletonList(server0));
+                server1.reconnect(Collections.singletonList(server2));
+
+                // Server0 and server2 has same number of link failure ie. 1, the one with
+                // larger endpoint should be marked as unresponsive.
+                String secondServerToKick = Collections.max(Arrays.asList(server0.getEndpoint(), server2.getEndpoint()));
+                waitForLayoutChange(layout -> layout.getUnresponsiveServers()
+                        .equals(Collections.singletonList(secondServerToKick)), corfuClient);
+
+                lastLayout = corfuClient.getLayout();
+
+                // Verify data path working fine
+                for (int i = 0; i < DEFAULT_TABLE_ITER; i++) {
+                    assertThat(table.get(String.valueOf(i))).isEqualTo(String.valueOf(i));
+                }
+
+                // 4th link failure rotation, reverse the rotating direction, disconnect between
+                // server1 and server2 and heal previous link failure between server1 and server2
+                server1.disconnect(Collections.singletonList(server2));
+                server2.reconnect(Collections.singletonList(server0));
+
+                // Wait for some time to ensure cluster stabilizes
+                // Server1 should stay in unresponsive set, no layout change
+                waitUninterruptibly(20);
```

Comment:
Maybe it's better to pass Duration instead of int?

Reply:
Done.